src/shogun/kernel/GaussianARDKernel.h

```diff
@@ -4,25 +4,60 @@
  * the Free Software Foundation; either version 3 of the License, or
  * (at your option) any later version.
  *
+ * Written (W) 2015 Wu Lin
  * Written (W) 2012 Jacob Walker
  *
  * Adapted from WeightedDegreeRBFKernel.h
  */
 
-#ifndef GAUSSIANARDKERNEL_H_
-#define GAUSSIANARDKERNEL_H_
+#ifndef _GAUSSIANARDKERNEL_H_
+#define _GAUSSIANARDKERNEL_H_
 
 #include <shogun/lib/config.h>
 
 #include <shogun/lib/common.h>
 #include <shogun/kernel/DotKernel.h>
-#include <shogun/features/DenseFeatures.h>
 #include <shogun/kernel/LinearARDKernel.h>
 
 namespace shogun
 {
 
-/** @brief Gaussian Kernel with Automatic Relevance Detection */
+/** @brief Gaussian Kernel with Automatic Relevance Detection computed
+ * on CDotFeatures.
+ *
+ * It is computed as
+ *
+ * \f[
+ * k({\bf x},{\bf y})= exp(-\frac{||{\bf x}-{\bf y}||}{\tau})
+ * \f]
+ *
+ * where \f$\tau\f$ is the kernel width.
+ *
+ * There are three variants based on \f$||\cdot||\f$. 
+ * The default case is
+ * \f$\sum_{i=1}^{p}{{[\lambda \times ({\bf x_i}-{\bf y_i})] }^2}\f$
+ * where \f$\lambda\f$ is a scalar and \f$p\f$ is # of features
+ * To use this case,  please call set_weights(\f$\lambda\f$),
```

Comment:
please also add convenience methods here and mention them.
Hide the actual method, make it protected